### PR TITLE
Summer jam - split commits

### PR DESCRIPTION
**Pull request by sornas**
Audio commit still needs a name but otherwise this is done.

### CODE REVIEW

File: src/input.rs [hunk 1]
```diff
@@ -197,6 +202,31 @@ where
...
+    pub fn text_input_update(&mut self, s: &mut String) -> bool {
+        let mut found_return = false;
+        for (keycode, keymod) in std::mem::take(&mut self.text_input_events) {
+            match keycode {
+                Keycode::Backspace => { s.pop(); }
+                Keycode::Escape => { s.clear(); found_return = true; }
+                Keycode::Return => found_return = true,
+                c => if let Some(mut c) = (c as i32).try_into().ok().and_then(char::from_u32) {
+                    //TODO(gu): Handle non-ascii better.
+                    if c.is_ascii_alphanumeric() || c.is_ascii_punctuation() || c == ' ' {
```
Comment (FredTheDino) [inline src/input.rs:218]: is there an `is ascii`?

File: src/renderer.rs [hunk 2]
```diff
@@ -386,9 +395,20 @@ impl Renderer {
...
+        let post_program = context
+            .new_shader_program::<VertexSemantics, (), PostShaderInterface>()
+            .from_strings(VS_POST_STR, None, None, FS_POST_STR)
+            .unwrap()
+            .ignore_warnings();
+
         let tex: Tex =
             Texture::new(context, SPRITE_SHEET_SIZE, 0, sampler).expect("failed to create texture");
 
+        // TODO(ed): Resize when we resize the window
```
Comment (FredTheDino) [inline src/renderer.rs:407]:
```suggestion
```
Is implemented

File: src/renderer.rs [hunk 3]
```diff
@@ -487,13 +524,18 @@ impl Renderer {
...
+        let tex = &mut self.tex;
+        let sprite_prog = &mut self.sprite_program;
+        let particle_prog = &mut self.particle_program;
+        let font = &mut self.font;
+
         let render = context
             .new_pipeline_gate()
             .pipeline(
-                &back_buffer,
+                &self.offscreen_buffer, // something_else
```
Comment (FredTheDino) [inline src/renderer.rs:535]:
```suggestion
                &self.offscreen_buffer,
```

File: src/fs_post.glsl [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            2.0 * texture(tex_col, v_uv + vec2( x,  0)).rgb +
+            2.0 * texture(tex_col, v_uv + vec2(-x,  0)).rgb +
+            2.0 * texture(tex_col, v_uv + vec2( 0, -y)).rgb +
+            2.0 * texture(tex_col, v_uv + vec2( 0,  y)).rgb +
+
+            1.0 * texture(tex_col, v_uv + vec2( x,  y)).rgb +
+            1.0 * texture(tex_col, v_uv + vec2(-x,  y)).rgb +
+            1.0 * texture(tex_col, v_uv + vec2(-x, -y)).rgb +
+            1.0 * texture(tex_col, v_uv + vec2( x, -y)).rgb
+    , 16.0) / 16.0;
```
Comment (FredTheDino) [inline src/fs_post.glsl:22]: Maybe this should be an identity shader? Or is it a good test?

**Review by FredTheDino — APPROVED** [review]

File: src/input.rs (hunk 1 above)
Comment (sornas) [reply src/input.rs:218]: Yes, but that also allows the control characters.

File: src/fs_post.glsl (hunk 4 above)
Comment (sornas) [reply src/fs_post.glsl:22]: I wonder if we'll get back to this. I think we should remove it from this PR and allow an engine user to use their own shader.

Comment (sornas) [reply src/fs_post.glsl]: https://github.com/sornas/lingon/tree/post-shaders contains the shader code, for future reference.

**Pull request merged**